server/src/instant/scripts/newsletter.clj

```diff
@@ -0,0 +1,152 @@
+    (add-error! to "ERROR" (.getMessage e)))
+  nil)
+
+(defn send! [{:keys [to title textBody htmlBody
+                     ;; alias refers to the email template identifer in postmark
+                     ;; In general you shouldn't need to provide any of these
+                     ;; optional params unless you're sending a custom email
+                     alias from reply-to]
+              :or {alias "instant-news"
+                   from "js@hey.instantdb.com"
+                   reply-to "founders@instantdb.com"}}]
+  (let [model {:title title
+               :textBody textBody
+               :htmlBody htmlBody}
+        body {:From from
+              :To to
+              :ReplyTo reply-to
+              :TemplateAlias alias
+              :TemplateModel model
+              :MessageStream "broadcast"
+              :TrackOpens true
+              :TrackLinks "HtmlAndText"
+              :InlineCss true}]
+    (if-not (config/postmark-send-enabled?)
+      (tracer/with-span! {:name "postmark/send-disabled"
+                          :attributes body}
+        (tracer/record-info!
+         {:name "postmark-disabled"
+          :attributes
+          {:msg
+           "Postmark is disabled, add postmark-token to config to enable"}}))
+      (tracer/with-span! {:name "postmark/send"
+                          :attributes body}
+        (try
+          (clj-http/post
+           "https://api.postmarkapp.com/email/withTemplate"
+           {:coerce :always
+            :as :json
+            :headers {"X-Postmark-Server-Token" (config/postmark-token)
+                      "Content-Type" "application/json"}
+            :body (->json body)})
+          (catch Exception e
+            (handle-email-error! e to)))))))
+
+(defn send-newsletter! [{:keys [title slug live?]}]
+  (let [emails (str/split-lines (read-file base-path emails-path))]
+    (if-not live?
+      (doseq [email emails]
+        (log/info "Simulating sending newsletter to" email))
+      (if @once
+        (throw (Exception. "Already sent newsletter! Set once atom to false to send again."))
+        (do
+          (reset! once true)
+          (doseq [email emails]
+            (let [htmlBody (read-file base-path (html-path slug))
+                  textBody (read-file base-path (txt-path slug))]
+              (send! {:to email
+                      :title title
+                      :htmlBody htmlBody
+                      :textBody textBody}))))))))
+
+(comment
+  ;; 1. Set your params
+  (def params {:title "Instant News - Oct 2024" :slug "oct2024"})
+
+  ;; 2. Send test email to yourself, verify looks good
+  (let [htmlBody (read-file base-path (html-path (:slug params)))
+        textBody (read-file base-path (txt-path (:slug params)))]
+    (send! (assoc params
+                  :to "joeaverbukh@gmail.com"
+                  :htmlBody htmlBody
+                  :textBody textBody)))
+
+  ;; 3. Add emails to server/resources/emails/emails.txt
```

Comment:
Since we're warming up right now I choose not to automate this part for this month. I dumped the list of users from the top page (so the most active are first) to a google sheet and am manually sending 100 daily emails this week, 200 daily the next week, etc.

Starting next month we can fetch this via a query. Will push an update then

Reply:
Love it!